docs/halibut-protocol.md

```diff
@@ -0,0 +1,55 @@
+# Halibut Protocol data exchange
+
+When invoking a remote method in both polling and listening mode, the same general process is followed:
+
+ - The Server sends a `Request` message containing the method to execute and data.
+ - The client executes the method and sends the result in a `Response` message.
+ - The client sends a `NEXT` control message.
+ - The Server semds a `Proceed` control message.
+ - The above 4 steps are repeated until the connection is terminated or control `END` messages are sent.
+
+
+## Listening client protocol data exchange
+
+![Listening client protocol data exchange](images/listeningprotocoldata.png)
+
+## Polling client protocol data exchange
+
+![Polling client protocol data exchange](images/pollingprotocoldata.png)
+
+## Request and Response message format
+
+Request and Response messages are sent and received in the same format with only the data within the BSON section changing to denote the type of message.
+
+![Request/Response message format](images/message-format.png)
```

Comment:
These images don't appear to be rendering all that well when I look at the [branch directly in GitHub](https://github.com/OctopusDeploy/Halibut/blob/luke/illustrate-the-halibut-protocol/docs/halibut-protocol.md), I wonder if they could do with a white background?

Reply:
I see them with a white background...

Reply:
Although for me they are blurry:

![image](https://user-images.githubusercontent.com/7076477/218373129-c60c87fa-5e2c-4d10-8dc6-a4b0135e5f9d.png)

Reply:
I think they have a transparent background, I'm using a dark theme in GitHub so the background appears dark and some of the words are hard to read.

<img width="889" alt="image" src="https://user-images.githubusercontent.com/2915931/218374990-a8b2b43a-2a8d-4018-8b8e-eddd23f31a94.png">

Reply:
oh boy

Reply:
If you re-export the pictures with a white background hopefully it should appear ok for both light and dark themes?

Reply:
try now?

Reply:
Yep looking good now, thanks